src/osp/tasks/execute_simple.cpp

```diff
@@ -25,58 +25,67 @@
 
 #include "execute_simple.h"
 
-#include <Corrade/Containers/ArrayViewStl.h>
-
 #include <algorithm>
-#include <cassert>
+#include <cstdint>
+
+using Corrade::Containers::ArrayView;
+using Corrade::Containers::StridedArrayView2D;
+using Corrade::Containers::arrayView;
 
 namespace osp
 {
 
-void task_enqueue(TaskTags const& tags, ExecutionContext &rExec, BitSpanConst_t const query)
+bool any_bits_match(BitSpanConst_t const lhs, BitSpanConst_t const rhs)
 {
-    std::size_t const tagIntSize = tags.tag_ints_per_task();
-    assert(query.size() == tagIntSize);
+    return ! std::equal(
+            std::cbegin(lhs), std::cend(lhs),
+            std::cbegin(rhs), std::cend(rhs),
+            [] (uint64_t const lhsInt, uint64_t const rhsInt)
+    {
```

Comment:
What's forcing the need for this lambda?

and why not use std::ranges::equal() ? i got the codebase compiling with C++20 didn't i?

Reply:
std::equal just happens to work as a way to run a functor element-wise against two ranges, and return true if they all return true. The original 'equals' intent of this function is misleading here, and I might as well use a for loop here.

I was not aware of std::ranges::equal when writing this.